src/main/java/org/springframework/data/elasticsearch/core/StreamQueries.java

```diff
@@ -56,8 +57,7 @@
 			@Override
 			public void close() {
 				try {
-					// Clear scroll on cluster only in case of error (cause elasticsearch auto clear scroll when it's done)
-					if (!finished && scrollId != null && currentHits != null && currentHits.hasNext()) {
+					if (scrollId != null) {
```

Comment:
can you please add `@Nullable` (package `org.springframework.lang` to the declaration of `scrollId`in line 51? Can't add a comment there

Reply:
The ScrollId shouldn't be zero, if you ask me. I actually just wanted to fix the problem, but now I think a complete rewrite of the `CloseableIterator' implementation would be better.

Reply:
The `ScrolledPage` explicitly has it defined as nullable. Then better make it non-nullable there and remove the checks here.

I was thinking if a rewrite is necessary as well yesterday evening, especially thinking about doing the close implicitly, when no more data is available. But I stepped back from that idea, as we explcitly return a `CloseablIterator` which makes it clear for me that it's the caller's responsibility to close this.

So I would not change this any further. What I need to do in a different ticket is to provide a ScrollPage implementation instead of the deprecated `AggregatedPageImpl` and make sure that all this scrolling stuff is package private. We made that for the methods in the Operations already, but I think the implementations are still public.
The scroll stuff should only be used internally for implementing the streaming.

Reply:
I removed `@Nullable` from `ScrolledPage` but left it in `AggregatedPageImpl` and made minor changes to the `CloseableIterator` in `StreamQueries`.